### DISCUSSION TITLE
Any plans to support embedding marimo WASM notebooks as React components?

### DISCUSSION THREAD

**Discussion by gvarnavi** [marimo-team/marimo#962] [Ideas]
First, let me start by saying I think marimo notebooks (and in-particular the WASM integration) is amazing!

The [docs](https://docs.marimo.io/guides/wasm.html#embedding) describe how to embed WASM notebooks using an iframe, which works well. I was wondering if there's plans to support more advanced embeddings, say into a React app similar to how [Observable Notebooks](https://observablehq.com/documentation/embeds/how-to-embed-a-notebook-in-a-react-app) can? Or is this already supported but not documented? Perhaps using the [@marimo-team/frontend](https://www.npmjs.com/package/@marimo-team/frontend) package?

If you're curious, my particular use-case is for interactive scientific presentations - see e.g. an example with embedded Observable notebooks [here](https://gvarnavides.com/interactive-electron-ptychography/slides/4dstem-intro) and jupyter-react components [here](https://gvarnavides.com/interactive-electron-ptychography/slides/phase-retrieval) (this is the part I'm hoping to replace with marimo WASM notebooks).

**Comment by akshayka** [comment]
Hey @gvarnavi, sorry for the late response. We haven't implemented embedding as React components yet.

Out of curiosity, why won't an iframe work for your use case?

**Comment by gvarnavi** [comment]
Hey @akshayka, the iframe embedding works well (and I plan to use that in the short term) - I was just hoping for deeper integration/ more control (e.g. css styling to match host website etc).
